notify/notify.go

```diff
@@ -729,3 +745,44 @@ func (n SetNotifiesStage) Exec(ctx context.Context, l log.Logger, alerts ...*typ
 
 	return ctx, alerts, n.nflog.Log(n.recv, gkey, firing, resolved)
 }
+
+type TimeMuteStage struct {
+	muteTimes map[string][]timeinterval.TimeInterval
+}
+
+func NewTimeMuteStage(mt map[string][]timeinterval.TimeInterval) *TimeMuteStage {
+	return &TimeMuteStage{mt}
+}
+
+// Exec implements the stage interface for TimeMuteStage
+// TimeMuteStage is responsible for muting alerts whose route is not in an active time
+func (mts TimeMuteStage) Exec(ctx context.Context, l log.Logger, alerts ...*types.Alert) (context.Context, []*types.Alert, error) {
+	muteTimeNames, ok := MuteTimeNames(ctx)
+	if !ok {
+		return ctx, alerts, nil
+	}
+	now, ok := Now(ctx)
+	if !ok {
+		return ctx, alerts, errors.New("missing now timestamp")
+	}
+
+	muted := false
+	for _, mtName := range muteTimeNames {
+		mt, ok := mts.muteTimes[mtName]
+		if !ok {
+			return ctx, alerts, errors.Errorf("mute time %s doesn't exist in config", mtName)
+		}
+		for _, ti := range mt {
+			if ti.ContainsTime(now) {
+				muted = true
```

Comment:
```suggestion
				muted = true
				break
```
You can break the outer loop here

Reply:
Thanks, good pickup. I'll fix this.